docs/dev-guide/plugin-dev.md

```diff
@@ -4,198 +4,91 @@ sidebarDepth: 3
+So, typical CLI plugin folder structure looks like the following:
 
 ```
 .
 ├── README.md
 ├── generator.js  # generator (optional)
-├── prompts.js    # prompts file (optional)
 ├── index.js      # service plugin
-└── package.json
-```
-
-### Service Plugin
-
-Service plugins are loaded automatically when a Service instance is created - i.e. every time the `vue-cli-service` command is invoked inside a project.
-
-Note the concept of a "service plugin" we are discussing here is narrower than that of a "CLI plugin", which is published as an npm package. The former only refers to a module that will be loaded by `@vue/cli-service` when it's initialized, and is usually a part of the latter.
-
-In addition, `@vue/cli-service`'s [built-in commands][commands] and [config modules][config] are also all implemented as service plugins.
-
-A service plugin should export a function which receives two arguments:
-
-- A [PluginAPI][plugin-api] instance
-
-- An object containing project local options specified in `vue.config.js`, or in the `"vue"` field in `package.json`.
-
-The API allows service plugins to extend/modify the internal webpack config for different environments and inject additional commands to `vue-cli-service`. Example:
-
-``` js
-module.exports = (api, projectOptions) => {
-  api.chainWebpack(webpackConfig => {
-    // modify webpack config with webpack-chain
-  })
-
-  api.configureWebpack(webpackConfig => {
-    // modify webpack config
-    // or return object to be merged with webpack-merge
-  })
-
-  api.registerCommand('test', args => {
-    // register `vue-cli-service test`
-  })
-}
-```
-
-#### Specifying Mode for Commands
-
-> Note: the way plugins set modes has been changed in beta.10.
-
-If a plugin-registered command needs to run in a specific default mode,
-the plugin needs to expose it via `module.exports.defaultModes` in the form
-of `{ [commandName]: mode }`:
-
-``` js
-module.exports = api => {
-  api.registerCommand('build', () => {
-    // ...
-  })
-}
-
-module.exports.defaultModes = {
-  build: 'production'
-}
+├── package.json
+├── prompts.js    # prompts file (optional)
+└── ui.js         # Vue UI integration (optional)
 ```
 
-This is because the command's expected mode needs to be known before loading environment variables, which in turn needs to happen before loading user options / applying the plugins.
+## Generator
 
-#### Resolving Webpack Config in Plugins
+A Generator part of the CLI plugin is usually needed when you want to extend your package with new dependencies, create new files in your project or edit existing ones.
 
-A plugin can retrieve the resolved webpack config by calling `api.resolveWebpackConfig()`. Every call generates a fresh webpack config which can be further mutated as needed:
+Inside the CLI plugin the generator should be placed inside a `generator.js` or `generator/index.js` file. It will be invoked in two possible scenarios:
```

Comment:
```suggestion
Inside the CLI plugin the generator should be placed in a `generator.js` or `generator/index.js` file. It will be invoked in two possible scenarios:
```